src/Admin.cpp

```diff
@@ -100,23 +143,118 @@ namespace geopm
 
     std::string Admin::whitelist(int cpuid)
     {
-        std::string result;
         if (cpuid == -1) {
-           if (m_msriog == nullptr) {
-               m_msriog = std::make_shared<MSRIOGroup>();
-           }
-           result = m_msriog->msr_whitelist();
+            cpuid = m_cpuid_local;
         }
-        else {
-            result = geopm::MSRIOGroup::msr_whitelist(cpuid);
+        return geopm::MSRIOGroup::msr_whitelist(cpuid);
+    }
+
+    std::vector<std::string> Admin::dup_keys(const std::map<std::string, std::string> &map_a,
+                                             const std::map<std::string, std::string> &map_b)
+    {
+        std::vector<std::string> result;
+        auto a_it = map_a.begin();
+        auto b_it = map_b.begin();
+        while (a_it != map_a.end() && b_it != map_b.end()) {
```

Comment:
This looks like it is incomplete if a has less keys than b, no?

Reply:
Maybe create 2 sets of keys (looping thru both maps independently) and return the intersection?

Reply:
I think it will work as long as keys are ordered with < (should be true for std::map).

Reply:
Should work to find all duplicates.  Do not have to keep looking once one of the maps has been exhausted.